### PR TITLE
Removed the publish metadata pipeline variable

### PR DESCRIPTION
**Pull request by dabasnidhi**
We will always push the metadata from the tasks as the saving part is now handled on server side project settings.
Also removed PublishMetadata task from make-options.json till the spec is final

### CODE REVIEW

File: Tasks/Common/docker-common-v2/pipelineutils.ts [hunk 1]
```diff
@@ -19,31 +19,25 @@ function addLabel(hostName: string, labelName: string, variableName: string, lab
     }
 }
 
-function addCommonLabels(hostName: string, labels: string[], addPipelineData?: boolean): void {
```
Comment (vithati) [inline Tasks/Common/docker-common-v2/pipelineutils.ts]: I think we can keep this. This is about adding default labels or not. Sync-up with Vishal.

Comment (dabasnidhi) [reply Tasks/Common/docker-common-v2/pipelineutils.ts]: Added back the check for pipeline metadata in docker task as it is used for adding labels.

File: Tasks/AzureFunctionOnKubernetesV0/task.json [hunk 2]
```diff
@@ -14,8 +14,8 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 0,
-        "Minor": 160,
-        "Patch": 8
+        "Minor": 161,
```
Comment (vithati) [inline Tasks/AzureFunctionOnKubernetesV0/task.json]: 162. other places also.

File: Tasks/ContainerStructureTestV0/task.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 0,
         "Minor": 1,
```
Comment (vithati) [inline Tasks/ContainerStructureTestV0/task.json]: 162 and patch version 0.

File: Tasks/DockerComposeV0/task.loc.json [hunk 4]
```diff
@@ -13,8 +13,8 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 160,
-    "Patch": 4
+    "Minor": 161,
```
Comment (vithati) [inline Tasks/DockerComposeV0/task.loc.json]: I think this file change not required.

**Pull request merged**